sync-core/src/main/java/com/cloudant/sync/query/IndexManager.java

```diff
@@ -376,4 +380,60 @@ protected SQLDatabase getDatabase() {
         return database;
     }
 
+    /**
+     * Check that the necessary settings to support text search are included
+     * in the SQLite compile options.
+     *
+     * @return text search enabled setting
+     */
+    protected static boolean ftsCompileOptionsAvailable(ExecutorService q, final SQLDatabase db) {
+        boolean ftsOptionsExist = false;
+        if (q != null && db != null) {
+            try {
+                ftsOptionsExist = q.submit(new Callable<Boolean>() {
+                    @Override
+                    public Boolean call() {
+                        int ftsOptionCount = 0;
+                        List<String> ftsCompileOptions = Arrays.asList("ENABLE_FTS3",
+                                                                       "ENABLE_FTS3_PARENTHESIS");
+                        String pragmaStatement = "PRAGMA compile_options;";
+                        Cursor compileOptions = null;
+                        try {
+                            compileOptions = db.rawQuery(pragmaStatement, new String[]{});
+                            while (compileOptions.moveToNext()) {
+                                String compileOption = compileOptions.getString(0);
+                                if (ftsCompileOptions.contains(compileOption)) {
+                                    ftsOptionCount = ftsOptionCount + 1;
+                                }
+                            }
+                        } catch (SQLException e) {
+                            String msg = "Failed to get a list of compile options from SQLite.";
+                            logger.log(Level.SEVERE, msg, e);
+                        } finally {
+                            DatabaseUtils.closeCursorQuietly(compileOptions);
+                        }
+                        return ftsOptionCount >= 2;
```

Comment:
You could also use something like:

```java
String compileOption = compileOptions.getString(0);
ftsCompileOptions.remove(complieOption);
// ...
return ftsCompileOptions.isEmpty();
```
Which more explicitly checks that we saw all the options we wanted, avoiding some odd query result where `ENABLE_FTS3` is included twice in the results for some reason.